### PR TITLE
[Enhancement] autovacuum without list tablet metadata

### PR DESCRIPTION
**Pull request by sduzh**
Executing autovacuum tasks no longer performs a list of tablet metadata. list tablet metadata is a costly operation, and when there is a large amount of tablet metadata (more than a few million), it takes a very long time for the results to be returned, affecting the speed of the vacuum.

Note that the autovacuum task still lists txn logs, which normally doesn't have too many files and shouldn't take too long, so it hasn't been optimized away for now.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 62 / 66 (93.94%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|62|66|93.94%|[127, 167, 168, 169]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 90 / 95 (94.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|90|95|94.74%|[130, 211, 212, 213, 214]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 93 / 102 (91.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|93|102|91.18%|[131, 216, 217, 218, 219, 220, 221, 222, 223]|

**Review by tracymacding — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 90 (92.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|83|90|92.22%|[130, 132, 211, 212, 213, 214, 215]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 90 (92.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|83|90|92.22%|[130, 132, 211, 212, 213, 214, 215]|

File: be/src/storage/lake/vacuum.cpp [hunk 1]
```diff
@@ -81,115 +87,149 @@ static Status delete_files(FileSystem* fs, const std::vector<std::string>& paths
...
-                    *vacuumed_file_size += file.size();
-                }
-                if (metadata->has_prev_garbage_version()) {
-                    garbage_version = metadata->prev_garbage_version();
+                // Reached here means that the tablet metadata that has been traversed was created after
+                // |grace_timestamp| and cannot be deleted.
+                // Now check if the current |version| was created before |grace_timestamp|.
+                ASSIGN_OR_RETURN(auto mtime, fs->get_file_modified_time(path));
+                TEST_SYNC_POINT_CALLBACK("collect_files_to_vacuum:get_file_modified_time", &mtime);
+                if (mtime < grace_timestamp) {
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/storage/lake/vacuum.cpp:146]: Yes, it's already under development: https://github.com/sduzh/starrocks/commit/17bf38925b3ad0bf83f2e8e5a84f5837c882748c

File: be/src/storage/lake/vacuum.cpp [hunk 2]
```diff
@@ -35,14 +33,18 @@
...
 namespace starrocks::lake {
 
-static bvar::LatencyRecorder g_del_file_latency("lake_vacuum_del_file");
+static bvar::LatencyRecorder g_del_file_latency("lake_vacuum_del_file"); // unit: us
 static bvar::Adder<uint64_t> g_del_fails("lake_vacuum_del_file_fails");
+static bvar::LatencyRecorder g_metadata_travel_latency("lake_vacuum_metadata_travel"); // unit: ms
+static bvar::LatencyRecorder g_txnlog_travel_latency("lake_vacuum_txnlog_travel");
 
 static Status delete_file(FileSystem* fs, const std::string& path) {
+#ifndef STARROCKS_VACUUM_DISABLE_FILE_DELETION
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/storage/lake/vacuum.cpp]: removed

File: be/src/storage/lake/vacuum.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/storage/lake/vacuum.cpp:138]: https://github.com/StarRocks/starrocks/pull/32424

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 90 (92.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/vacuum.cpp|83|90|92.22%|[122, 124, 203, 204, 205, 206, 207]|

**Review by wyb — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**